applications/CoSimulationApplication/python_scripts/development_helpers/external_field_solver.py

```diff
@@ -0,0 +1,184 @@
+            if coupling_interface_settings.Has("receive_interface"):
+                receive_interface = coupling_interface_settings["receive_interface"].GetBool()
+                operation = "Importing"
+
+            self.__CustomPrint(2, colors.cyan(operation)+' coupling-interface "{}" on ModelPart "{}"'.format(comm_name, sub_model_part_name))
+
+            if not self.dry_run:
+                if receive_interface:
+                    # use an aux-modelpart, which will not be used again
+                    KratosCoSim.EMPIRE_API.EMPIRE_API_recvMesh(self.model.CreateModelPart(sub_model_part_name), comm_name)
+                    # TODO maybe restructure and do all receives after all sends
+                else:
+                    KratosCoSim.EMPIRE_API.EMPIRE_API_sendMesh(self.model["ExtSolver."+sub_model_part_name], comm_name)
+            else:
+                self.__CustomPrint(2, colors.magenta('... skipped'))
+
+            self.__CustomPrint(1, colors.cyan("Finished "+operation) + " of CouplingInterfaces")
+
+        # time loop
+        self.__CustomPrint(1, "Starting Solution Loop")
+        for i in range(self.settings["num_steps"].GetInt()):
+            print() # newline
+            self.__CustomPrint(1, colors.bold('Step: {}/{}'.format(i+1, self.settings["num_steps"].GetInt())))
+            self.SolveSolutionStep()
+        self.__CustomPrint(1, "Finished")
+
+    def SolveSolutionStep(self):
+        num_coupling_interfaces = self.settings["coupling_interfaces"].size()
+
+        self.__CustomPrint(1, colors.green("Starting import") + " of CouplingInterfaceData ...")
+
+        for i in range(num_coupling_interfaces):
+            coupling_interface_settings = self.settings["coupling_interfaces"][i]
+            if not coupling_interface_settings.Has("data_field_recv"):
+                continue
+            sub_model_part_name = coupling_interface_settings["sub_model_part_name"].GetString()
+            data_field_settings = coupling_interface_settings["data_field_recv"]
+
+            data_field_name = data_field_settings["data_field_name"].GetString()
+
+            self.__CustomPrint(2, colors.green('Importing')+' data-field "{}" on ModelPart "{}"'.format(data_field_name, sub_model_part_name))
+
+            variables = GenerateVariableListFromInput(data_field_settings["variables"])
+
+            if not self.dry_run:
+                KratosCoSim.EMPIRE_API.EMPIRE_API_recvDataField(self.model["ExtSolver."+sub_model_part_name], data_field_name, *variables)
+            else:
+                self.__CustomPrint(2, colors.magenta('... skipped'))
+
+        self.__CustomPrint(1, colors.green("Finished import") + " of CouplingInterfaceData")
+
+        print() # newline
+        self.__CustomPrint(1, colors.blue("Solving ..."))
+        time.sleep(self.solving_time)
+        self.__CustomPrint(2, "Solving took {} [sec]".format(self.solving_time))
+        print() # newline
+        # TODO implement random values ... ?
```

Comment:
Yes change the fields based on the current time step (not random) so that it can be used in tests :-)

Reply:
will do that in a future PR